test/helpers/helpers.go

```diff
@@ -373,3 +391,168 @@ func CheckCscSuccessfulDeletion(client test.FrameworkClient, cscName, namespace,
+			Kind: v2.CatalogSourceConfigKind,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels:    labels,
+		},
+		Spec: v2.CatalogSourceConfigSpec{
+			TargetNamespace: namespace,
+			Packages:        "",
+		},
+	}
+}
+
+// CreateInstalledCscDefinition returns a newly built CatalogSourceConfig
+func CreateInstalledCscDefinition(namespace string) *v2.CatalogSourceConfig {
+	name := TestInstalledCscPublisherName + namespace
+	return &v2.CatalogSourceConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: fmt.Sprintf("%s/%s",
+				v1.SchemeGroupVersion.Group, v1.SchemeGroupVersion.Version),
+			Kind: v2.CatalogSourceConfigKind,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: v2.CatalogSourceConfigSpec{
+			TargetNamespace: namespace,
+			Packages:        "",
+		},
+	}
+}
+
+// CreateUISubscriptionDefinition returns a newly built Subscription with the labels
+// `csc-owner-name` and `csc-owner-namespace` and the expected name
+func CreateUISubscriptionDefinition(name, namespace string, isCreatedByUI bool) *olm.Subscription {
+	labels := make(map[string]string)
+	specSource := TestInstalledCscPublisherName + namespace
+	if isCreatedByUI {
+		labels[builders.CscOwnerNameLabel] = specSource
+		labels[builders.CscOwnerNamespaceLabel] = namespace
+	}
+
+	return &olm.Subscription{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: fmt.Sprintf("%s/%s",
+				olm.SchemeGroupVersion.Group, olm.SchemeGroupVersion.Version),
+			Kind: olm.SubscriptionKind,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels:    labels,
+		},
+		Spec: &olm.SubscriptionSpec{
+			CatalogSource:          specSource,
+			CatalogSourceNamespace: namespace,
+			Channel:                "alpha",
+		},
+	}
+}
+
+// CreateUserSubscriptionDefinition returns a newly built Subscription
+func CreateUserSubscriptionDefinition(name, namespace string) *olm.Subscription {
+	specSource := TestInstalledCscPublisherName + namespace
+	return &olm.Subscription{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: fmt.Sprintf("%s/%s",
+				olm.SchemeGroupVersion.Group, olm.SchemeGroupVersion.Version),
+			Kind: olm.SubscriptionKind,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: &olm.SubscriptionSpec{
+			CatalogSource:          specSource,
+			CatalogSourceNamespace: namespace,
+			Channel:                "alpha",
+		},
+	}
+}
+
+// CheckSubscriptionNotUpdated checks that a user created subscription
+// was not updated during migration.
+func CheckSubscriptionNotUpdated(client test.FrameworkClient, name, namespace string) error {
+	subscription := &olm.Subscription{}
+	specSource := TestInstalledCscPublisherName + namespace
+	err := client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, subscription)
+	if err != nil {
+		return err
+	}
+	if subscription.Spec.CatalogSource != specSource {
```

Comment:
Is it reasonable to check resourceversion here? Or will something else potentially (validly) update the subscription?